File: Semestrul 4/MPP/Proiecte/java_gui/src/main/java/map/service/VoluntarService.java
package map.service;

import map.domain.Voluntar;
import map.repository.database.VoluntarDBRepo;
import map.repository.database.VoluntarRepo0;
import org.mindrot.jbcrypt.BCrypt;

public class VoluntarService{

    private VoluntarRepo0 voluntarRepo;

    public VoluntarService(VoluntarDBRepo voluntarRepo) {
        this.voluntarRepo = voluntarRepo;
    }

    public Voluntar login(String username, String parola) throws Exception {
        Voluntar voluntar = voluntarRepo.findByUsername(username);
        if (voluntar != null && BCrypt.checkpw(parola, voluntar.getPassword()))
            return voluntar;
        throw new Exception("Date de autentificare incorecte!");
    }

}

File: Semestrul 4/MPP/Proiecte/c#_gui/service/VoluntarService.cs
using System;
using System.Threading.Tasks;
using csharp_avalonia.domain;
using csharp_avalonia.repository;

namespace csharp_avalonia.service;

public interface IVoluntarService
{
    Task<Voluntar> Login(string username, string password);
}

public class VoluntarService
{
    private readonly VoluntarRepo0 voluntarRepo;

    public VoluntarService(VoluntarRepo voluntarRepo1)
    {
        voluntarRepo = voluntarRepo1;
    }

    public Voluntar Login(string username, string parola)
    {
        Voluntar voluntar = voluntarRepo.Login(username, parola);
        if (voluntar != null)
            return voluntar;
        throw new System.Exception("Date de autentificare incorecte!");
    }
}
